dbt/adapters/athena/session.py

```diff
@@ -10,3 +22,244 @@ def get_boto3_session(connection: Connection) -> boto3.session.Session:
         region_name=connection.credentials.region_name,
         profile_name=connection.credentials.aws_profile_name,
     )
+
+
+class AthenaSparkSessionManager:
```

Comment:
@Avinash-1394 can the same session be reused across multiple models?
Let' image that I have 4 models with tag `spark`, if I submit `dbt run -s tag:spark`
will the same session be reused across the models?

Reply:
In this example why 2 sessions are created/used and not one? Where is the logic that decide how many sessions we create? It could be a bit expensive having too many sessions.

Reply:
Yes that's correct. In case that is not specified I do use the `DEFAULT_THREADS` variable declared in `constants.py`. Let me know if it would be better to use a separate variable like `DEFAULT_SPARK_THREADS`